Adding libpyvinyl and mcstasscript

tkittel opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

mads-bertelsen commented:
I confirm that I want to be added as recipe-maintainer!

tkittel commented:
@conda-forge/help-python, ready for review!

ocefpaf commented on recipes/libpyvinyl/meta.yaml:
```diff
@@ -0,0 +1,73 @@
+  url: https://github.com/PaNOSC-ViNYL/libpyvinyl/archive/v{{ version }}.tar.gz
+  sha256: 1e83f89b3b702d6b527bba6a2efd64b4c76cad23ad8a8c305ad356381bcd4e93
+
+  #url: https://pypi.io/packages/source/l/libpyvinyl/libpyvinyl-{{ version }}.tar.gz
+  ##NB: get sha256 from https://pypi.org/project/libpyvinyl/#files
+  #sha256: 5031eacda47519734adf0cea21c684bb368659f08f8df48a698b56fb14c837e3
```
Why not the PyPI tarball? Also, can you remove these commented out lines?

tkittel replied:
Because of an issue in the upstream project (we did try the pypi source first), meaning that the pypi source dists are broken. This has been reported upstream, but in the meantime the github tarballs are needed as a workaround.

ocefpaf commented on recipes/libpyvinyl/meta.yaml:
```diff
@@ -0,0 +1,73 @@
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

ocefpaf commented on recipes/libpyvinyl/meta.yaml:
```diff
@@ -0,0 +1,73 @@
+    - libpyvinyl
+  commands:
+    - pip check
+    - pytest .  # [unix]
```
This is noarch, you don't need this selector. However, if there is a significant failure on Windows we need to know so we don't ship a broken package there. Noarch packages are built on linux and ship to all platforms. 

```suggestion
    - pytest .
```

tkittel replied:
Yes, this is just to make the windows builds pass since a few of the pytests were giving false positives there (essentially the upstream project needs to fix their unit tests to also run on windows, a few tests were using wrong path separators). But the package does work on windows.

ocefpaf commented on recipes/mcstasscript/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+
+source:
+  url: https://pypi.io/packages/source/m/mcstasscript/mcstasscript-{{ version }}.tar.gz
+  #NB: get sha256 from https://pypi.org/project/mcstasscript/#files
```
```suggestion
```

ocefpaf replied:
PS: The bot will take care of that for you in the feedstock.

ocefpaf commented on recipes/mcstasscript/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
/
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

tkittel commented:
@ocefpaf thanks a lot for the review! I basically just accepted all of your suggestions, and removed one extra comment like the one you wanted cleaned up.

The windows build is now failing, since I removed the selector you requested. But the package actually DOES work on windows, the issue is merely that the upstream project has a bug in their tests which gives a false positive failure on windows (some hardcoded path separator instead of using pathlib, but just in the testing code). Another upstream bug is the reason for using github tar-balls, as I also explained. Both issues have been reported upstream, as has the issue of a pinned dependency.

But in any case, I hope you are happy to accept this for conda-forge now? The upstream libpyvinyl issues are not specific to the conda-forge recipe, do not break the conda-forge package, and will be followed up upon by @mads-bertelsen .

ocefpaf commented:
> upstream project has a bug in their tests which gives a false positive failure on windows (some hardcoded path separator instead of using pathlib

That is not a blocker for is to merge here. Thanks for checking!
